serde/src/private/de.rs

```diff
@@ -2587,17 +2587,43 @@ where
 }
 
 #[cfg(any(feature = "std", feature = "alloc"))]
-pub struct FlatMapDeserializer<'a, 'de: 'a, E>(
-    pub &'a mut Vec<Option<(Content<'de>, Content<'de>)>>,
-    pub PhantomData<E>,
-);
+#[cfg_attr(any(feature = "std", feature = "alloc"), derive(Debug))]
+pub enum FlatMapDeserializerError {
+    Message(String),
+    NoVariantFoundInFlattenedData(&'static str),
+}
```

Comment:
Isn't that possible to keep original error instead of stringify it?
```suggestion
pub enum FlatMapDeserializerError<E> {
    Inner(E),
    NoVariantFoundInFlattenedData(&'static str),
}
```